apps/api/core/logs.py

```
@@ -1,7 +1,92 @@
+    uvicorn_error_logger.propagate = False
+
+    # Also configure the root logger to catch logs from other libraries
+    # and format them as JSON, but only if they are not uvicorn logs
+    # (to avoid double logging if uvicorn itself logs to root)
+    root_logger = logging.getLogger()
+    root_logger.setLevel(log_level)
+    # Remove existing handlers from root logger to avoid conflicts or default formatting
+    root_logger.handlers = []
+    root_logger.addHandler(stream_handler) # Add our JSON handler
+
+    logger.info(f"Logging setup complete for '{app_name}' with level {logging.getLevelName(log_level)}")
```

Ruff (0.11.9)

Logging statement uses f-string

(G004)